src/event_object_movement.c

```diff
@@ -1734,7 +1734,8 @@ static u16 PackGraphicsId(const struct ObjectEventTemplate *template)
         if (template->script && template->script[0] == 0x7d)
         {
             form = T1_READ_16(&template->script[2]);
-            form = (form >> 10) & 0x1F;
+            //form = (form >> 10) & 0x1F;
```

Comment:
any reason this isn't just removed?

Reply:
Showing the previous state of the code.
I'm not sure if this part of the code does what I think it does.

Reply:
Can we not check the previous state through git or this pull request?

Reply:
Yes, but it's also the code that's meant to stick out because I'm not sure about it.
It can be removed on Github when someone who knows more agrees that the new line is correct.

Reply:
Huh. `0x7d` is [`bufferspeciesname`](https://github.com/pret/pokeemerald/blob/master/asm/macros/event.inc#L1042), and `T1_READ_16(&template->script[2])` corresponds to the [`species` argument](https://github.com/pret/pokeemerald/blob/master/asm/macros/event.inc#L1044).

So I suppose what we're saying here is that if the script starts with `bufferspeciesname _, X` then `form` will be stored in the top 5 bits of `X`.

Ah, there's a comment just above which explains what I said, lol!:
```c
    // set form based on template's script,
    // if first command is bufferspeciesname
```
I think this change is needed because our species have 12 bits. I think for sanity's sake `form = template->trainerRange_berryTreeId & 0x1F;` should probably also use a `0x0F` mask because there's only 4 bits available for `form` in the graphics ID. There should probably be a constant for that mask to make it easier to understand what's going on? `OBJ_EVENT_GFX_FORM_MASK` or something.

I have to say I'm quite confused about the purpose of this code. Is the idea that maybe you'd have an Unown's graphics ID set to the base form, but then in the script you could specify a specific Unown form for it to be displayed as (I guess there's a macro somewhere that does `(species) | ((form) << 10)`?)

Reply:
Ah, also the shift in this should be by `OBJ_EVENT_GFX_SPECIES_BITS` rather than by `10` (again because of our species having 12 bits).